lib/buildkite/test_collector/test_links_plugin/reporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Buildkite::TestCollector::TestLinksPlugin
+  class Reporter
+    RSpec::Core::Formatters.register self, :dump_failures
+
+    def initialize(output)
+      @output = output
+    end
+
+    def dump_failures(notification)
+      # Do not display summary if no failed examples
+      return false unless notification.failed_examples.length.positive?
+
+      # Check if a Test Analytics token is set
+      return false unless Buildkite::TestCollector.api_token
+
+      # If the suite_url fails to be generated, then we are unable to create the test links
+      return false unless (url = suite_url)
```

Comment:
`unless` is used to evaluate a boolean condition, however you're doing an assignment here, so this condition will always evaluate to true. 

What are you intending to check here? The `suite_url` method doesn't return a boolean. I think you're trying to check if a `suite_url` was able to be fetched - but the way this method works currently doesn't allow you to do that? I've also got a comment later on about this method, I think in general it needs so redesigning, let me know if you wanna pair!

Reply:
The `suite_url` on line 35 will return `nil` if the feature flag is off, so this is checking for that. It is being assigned to `url` and if `url` is nil it will return false. Maybe I am just missing the check here for if `suite_url` was unable to be fetched?

Reply:
Also my comment above this line is a bit misleading!

Reply:
Yeah I think this bit is a bit confusing! It might make more sense once you rearchitect how the suite url is fetched? Give me a bell if you wanna pair :)